From 8e8f32367dc1e8e163ca125326949bb8a9e8c262 Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Fri, 6 Sep 2024 10:03:37 -0400
Subject: [PATCH 5/5] fix glibc headers

Something about the glibc headers makes it so that if things are included in
the wrong order, you can get errors like the following:

```
In file included from work/cctools/ld64/src/ld/libcodedirectory.c:42:
In file included from $HOST/include/dispatch/dispatch.h:44:
In file included from work/cctools/include/foreign/fcntl.h:1:
In file included from $BUILD/x86_64-conda-linux-gnu/sysroot/usr/include/fcntl.h:77:
$BUILD/x86_64-conda-linux-gnu/sysroot/usr/include/bits/stat.h:106:31: error: expected member name or ';' after declaration specifiers
  106 |     __syscall_slong_t __unused[3];
```

We can avoid these issues by making sure that certain headers are included
sufficiently early.
---
 cctools/ld64/src/ld/libcodedirectory.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cctools/ld64/src/ld/libcodedirectory.c b/cctools/ld64/src/ld/libcodedirectory.c
index 800241b..f3d3601 100644
--- a/cctools/ld64/src/ld/libcodedirectory.c
+++ b/cctools/ld64/src/ld/libcodedirectory.c
@@ -12,6 +12,7 @@
 #include <stdio.h>
 #include <stdarg.h>
 #include <errno.h>
+#include <fcntl.h>
 #include "3rd/helper.h" // ld64-port
 
 #include <arpa/inet.h>
